salt/fileclient.py

```diff
@@ -142,12 +143,21 @@ def _cache_loc(self, path, saltenv='base', cachedir=None):
                                     path)
         destdir = os.path.dirname(dest)
         cumask = os.umask(63)
-        if not os.path.isdir(destdir):
-            # remove destdir if it is a regular file to avoid an OSError when
-            # running os.makedirs below
-            if os.path.isfile(destdir):
-                os.remove(destdir)
+
+        # remove destdir if it is a regular file to avoid an OSError when
+        # running os.makedirs below
+        if os.path.isfile(destdir):
+            os.remove(destdir)
+
+        # ensure destdir exists
+        try:
             os.makedirs(destdir)
+        except OSError as exc:
+            if exc.errno == errno.EEXIST:  # already there, move on
+                pass
+            else:
+                raise
```

Comment:
@moio you're here asking advice whether you should or not write:
```python
if trouble:
  pass
else:
  do_something()
```
We normally write:
```python
if not trouble:
  do_something()
```
I fail to understand why these two extra lines needs to be there as a noise in the code.